flow/record/base.py

```diff
@@ -201,7 +201,27 @@ def _replace(self, **kwds):
         return result
 
     def __hash__(self) -> int:
-        return hash(self._pack(excluded_fields=IGNORE_FIELDS_FOR_COMPARISON))
+        desc_identifier, values = self._pack(excluded_fields=IGNORE_FIELDS_FOR_COMPARISON)
+        if not any((isinstance(value, list) for value in values)):
+            return hash((desc_identifier, values))
+
+        # Lists have to be converted to tuples to be able to hash them
+        record_values = []
+        for value in values:
+            if not isinstance(value, list):
+                record_values.append(value)
+                continue
+            list_values = []
+            for list_value in value:
+                if isinstance(list_value, dict):
+                    # List values that are dicts must be converted to tuples
+                    dict_as_tuple = ((k, v) for k, v in list_value.items())
+                    list_values.append(dict_as_tuple)
```

Comment:
Can be simplified to:

```suggestion
                    list_values.append(tuple(list_value.items()))
```